### PR TITLE
[TIDOC-2151] Update docs

### PR DESCRIPTION
**Pull request by bhatfield**

### CODE REVIEW

**Comment by bhatfield** [comment]
@ashcoding @cheekiatng Could one of you review?

File: apidoc/Facebook.yml [hunk 1]
```diff
@@ -227,134 +380,249 @@ description: |
...
+            those marked not-creation or read-only.
+
+            **Note:** You must set the `lifecycleContainer` property.
+        type: Dictionary<Titanium.Proxy>
+    returns:
+        type: Titanium.Proxy
+    since: 4.0.0
+    platforms: [android]
+
+  - name: dialog
```
Comment (ashcoding) [inline apidoc/Facebook.yml:433]: The new module doesn't have "dialog". This only existed in the older module. 
In the new module, instead you use presentSendRequestDialog, presentShareDialog, presentWebShareDialog.

File: apidoc/Facebook.yml [hunk 2]
```diff
@@ -227,134 +380,249 @@ description: |
...
+        summary: |
+            Array of additional permissions to request. For a complete list of permissions, see the
+            [official Facebook Permissions Reference](http://developers.facebook.com/docs/reference/api/permissions/)
+        type: Array<String>
 
-        For a list of dialogs, parameters, and response formats, see the 
-        [official documentation for Facebook Dialogs](https://developers.facebook.com/docs/android/send-requests).  
+      - name: audience
+        summary: |
+            The extent of the visibility write permissions will have. The value of audience should be one of "me", "friends", or "everyone"
```
Comment (ashcoding) [inline apidoc/Facebook.yml:580]: I think this should be the properties listed below. eg AUDIENCE_ONLY_ME. This is not type String.

File: apidoc/Facebook.yml [hunk 3]
```diff
@@ -227,134 +380,249 @@ description: |
...
+    description: |
+        Facebook now grants total control over granted permissions, and if the user modified the permissions
+        outside of your app your cached token may not be updated.
+
+        Listen for the <Modules.Facebook.tokenUpdated> event to be notified if the attempt was
+        successful.
+    since: 4.0.0
+
+  - name: request
+    summary: Makes a request to the legacy Facebook REST API.
```
Comment (ashcoding) [inline apidoc/Facebook.yml:598]: This is already removed and doesn't exist. Maybe that information can be added to this.

File: apidoc/Facebook.yml [hunk 4]
```diff
@@ -517,51 +774,174 @@ events:
...
   - name: AUDIENCE_EVERYONE
+    summary: Published content is visible to all Facebook users.
+    description: |
+        Use to set the default audience with either [LoginButton.audience](Modules.Facebook.LoginButton.audience)
+        or <Modules.Facebook.requestNewPublishPermissions>.
+    type: Number
+    permission: read-only
+    since: 4.0.0
+
+  - name: BUTTON_STYLE_NORMAL
```
Comment (ashcoding) [inline apidoc/Facebook.yml:821]: This doesn't exist for the new button.

File: apidoc/Facebook.yml [hunk 5]
```diff
@@ -517,51 +774,174 @@ events:
...
-        the default audience.
+        Use with [LoginButton.style](Modules.Facebook.LoginButton.style) to specify
+        the default login button reading "Connect" or "Login".
+    deprecated:
+        since: 4.0.0
+        removed: 4.0.0
     type: Number
     permission: read-only
 
+  - name: BUTTON_STYLE_WIDE
```
Comment (ashcoding) [inline apidoc/Facebook.yml:831]: This doesn't exist for the new button.

File: apidoc/Facebook.yml [hunk 6]
```diff
@@ -517,51 +774,174 @@ events:
...
+    type: Boolean
+    since: 4.0.0
+    permission: read-only
+
   - name: expirationDate
     summary: Time at which the `accessToken` expires.
     type: Date
     permission: read-only
 
+  - name: forceDialogAuth
```
Comment (ashcoding) [inline apidoc/Facebook.yml:900]: This doesn't exist in iOS any more if I recall correctly.

**Comment by ashcoding** [comment]
Review completed. Please address comments. Thank you!

File: apidoc/Facebook.yml (hunk 1 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:433]: This is marked as deprecated and removed.  Still need to keep it around for people using older versions of the module that don't want to upgrade right away.

File: apidoc/Facebook.yml (hunk 5 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:831]: Marked as deprecated and removed.

File: apidoc/Facebook.yml (hunk 6 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:900]: Marked as deprecated and removed.

File: apidoc/Facebook.yml (hunk 4 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:821]: Marked as deprecated and removed.

File: apidoc/Facebook.yml (hunk 3 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:598]: Marked as deprecated and removed.

File: apidoc/Facebook.yml (hunk 2 above)
Comment (bhatfield) [inline apidoc/Facebook.yml:580]: The method is marked as deprecated and removed.  This was the behavior for the method.  There weren't constants when this method was introduced in 3.1.0.

File: apidoc/Facebook.yml (hunk 1 above)
Comment (ashcoding) [inline apidoc/Facebook.yml:433]: Ah. Ok.

**Comment by ashcoding** [comment]
Ok. All is good.
Reviewed.

**Comment by bhatfield** [comment]
@ashcoding I readded the old APIs since I don't think most developers will be updating to the new module right away.  We need to keep the old APIs for reference.  They are marked as deprecated and removed and will only show up in the docs if people enable the `removed` filter in the docs website.

**Comment by ashcoding** [comment]
@bhatfield Got it. All is good then.
Validated and Reviewed. **Accepted**

**Pull request merged**